File: backend/app/domain/entities/project.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class Project:
    id: int
    title: str
    slug: str
    summary: str
    description: str
    cover_image_url: str | None
    gallery_image_urls: list[str]
    tech_stack: list[str]
    live_url: str | None
    github_url: str | None
    is_featured: bool
    sort_order: int
    is_published: bool
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/project.ts
export type Project = {
  id: number;
  title: string;
  slug: string;
  summary: string;
  description: string;
  cover_image_url: string | null;
  gallery_image_urls: string[];
  tech_stack: string[];
  live_url: string | null;
  github_url: string | null;
  is_featured: boolean;
  sort_order: number;
  is_published: boolean;
  created_at: string;
  updated_at: string;
};
